src/app/examples/examples.state.ts

```diff
@@ -12,11 +14,15 @@ export const reducers: ActionReducerMap<ExamplesState> = {
   stockMarket: stockMarketReducer
 };
 
-export const selectExamples = createFeatureSelector<any, ExamplesState>(
+export const selectExamples = createFeatureSelector<State, ExamplesState>(
   FEATURE_NAME
 );
 
 export interface ExamplesState {
   todos: TodosState;
   stockMarket: StockMarketState;
 }
+
+export interface State extends AppState {
```

Comment:
I would like to understand more about the tradeoff of this solution. This seems to lead to "re-creating" application state in every lazy loaded module. What would happen if we defined `examples` property in the original `AppState` ? Maybe it leads to circular deps...

Or is this the official ngrx way? If so maybe a bit change of naming ? Like `ExtendedAppState` or something that signifies what we're dealing with.

Reply:
I followed the NgRx example-app. 
But in our case this "ExtendedState" has little to no benefit.
If you would ever need something from the `AppState` in your feature module's selectors, this would be the way to make your feature module "smart" enough to know what's inside the AppState's model. 

We can remove this, the only impact it has is that we won't be able to safely type our `createFeatureSelector`, this would look as the following:

```
export const selectExamples = createFeatureSelector<ExamplesState>(FEATURE_NAME);
```

Personally I don't really have an opinion on this, this may help beginners, or this could be confusing them.

Reply:
I see, thank you for the explanation! Let's keep it! I can imagine use cases when we might need `auth` or `settings` stuff from store also in other feature modules. I will try to come up with such example and implement it so that it is used for real in our example app. 👍